reporter/otlp_reporter.go

```diff
@@ -723,15 +727,31 @@ func createFunctionEntry(funcMap map[funcInfo]uint64,
 
 // addProfileAttributes adds attributes to Profile.attribute_table and returns
 // the indices to these attributes.
-func addProfileAttributes(profile *profiles.Profile,
-	attributes []attrKeyValue, attributeMap map[string]uint64) []uint64 {
+func addProfileAttributes[T string | int64](profile *profiles.Profile,
+	attributes []attrKeyValue[T], attributeMap map[string]uint64) []uint64 {
 	indices := make([]uint64, 0, len(attributes))
 
-	addAttr := func(attr attrKeyValue) {
-		if attr.value == "" {
+	addAttr := func(attr attrKeyValue[T]) {
+		var attributeCompositeKey string
+		var attributeValue common.AnyValue
+
+		switch val := any(attr.value).(type) {
+		case string:
+			if val == "" {
+				return
+			}
+			attributeCompositeKey = attr.key + "_" + val
+			attributeValue = common.AnyValue{Value: &common.AnyValue_StringValue{StringValue: val}}
+		case int64:
+			if val == 0 {
+				return
+			}
+			attributeCompositeKey = attr.key + "_" + strconv.Itoa(int(val))
+			attributeValue = common.AnyValue{Value: &common.AnyValue_IntValue{IntValue: val}}
+		default:
 			return
```

Comment:
We should log an error here. Reaching this line is not expected and should not happen.